src/pheval/analyse/rank_stats.py

```diff
@@ -158,6 +160,55 @@ def return_mean_reciprocal_rank(self) -> float:
         else:
             return self.mean_reciprocal_rank()
 
+    @staticmethod
+    def _calculate_average_precision(
+        number_of_relevant_entities_at_k: int, precision_at_k: float
+    ) -> float:
+        """
+        Calculate the Average Precision at k.
+
+        Average Precision at k (AP@k) is a metric used to evaluate the precision of a ranked retrieval system.
+        It measures the precision at each relevant position up to k and takes the average.
+
+        Args:
+            number_of_relevant_entities_at_k (int): The count of relevant entities in the top-k predictions.
+            precision_at_k (float): The precision at k - the sum of the precision values at each relevant position.
+
+        Returns:
+            float: The Average Precision at k, ranging from 0.0 to 1.0.
+                   A higher value indicates better precision in the top-k predictions.
+        """
+        try:
+            return (1 / number_of_relevant_entities_at_k) * precision_at_k
+        except ZeroDivisionError:
+            return 0
+
+    def mean_average_precision_at_k(self, k: int) -> float:
+        """
+        Calculate the Mean Average Precision at k.
```

Comment:
Arent the words "Mean" and "average" the same? Above you just say "average" in the private method. Maybe "median average" is a thing, I wouldn't know :D

Reply:
Yep, mean and average are the same, it took me a while to understand the calculation because of its name. The private method first calculates the average precision for a single case in the corpora for the run and then this `mean_average_precision_at_k` calculates the mean for all cases in the corpora for a single run

Reply:
Sure, in this case, why not just name `mean_precision_at_k` - I am not bothered either way :P

Reply:
Keep it as MAP as this is what its commonly called.